lib/StripeResource.js

```diff
@@ -239,30 +239,46 @@ StripeResource.prototype = {
 
   _request: function(method, path, data, auth, options, callback) {
     var self = this;
+    var requestProcessor;
     var requestData;
 
     if (self.requestDataProcessor) {
-      requestData = self.requestDataProcessor(method, data, options.headers);
+      requestProcessor = self.requestDataProcessor(method, data);
     } else {
       requestData = utils.stringifyRequestData(data || {});
     }
 
     var apiVersion = this._stripe.getApiField('version');
 
-    var headers = self._defaultHeaders(auth, requestData.length, apiVersion);
+    if (requestProcessor) {
+      requestProcessor(method, data, options.headers, function(err, buffer) {
+        if (err) {
+          var handleError = self._errorHandler({}, callback);
```

Comment:
I think this one is probably better handled with a different kind of error, rather than overloading the connection error, since this isn't a connection (or other Stripe) error; (i presume) it'll be something related to the Buffer itself, so we should probably expose it as such.

Reply:
ok, cool - makes sense. Looking back now, it was kind of a 'cop out' to overload the connection error. I'll resolve to update and expose the error as such.